src/services/fastify.ts

```diff
@@ -20,6 +21,26 @@ const DEFAULT_REQUEST_ID_HEADER = 'X-Request-Id';
 const SERVER_PORT = 8080;
 const SERVER_HOST = '0.0.0.0';
 
+function getOauth2PluginOptions(): FastifyAuth0VerifyOptions {
+  const audience = env.get('OAUTH2_TOKEN_AUDIENCE').required().asString();
+  const domain = env.get('OAUTH2_JWKS_URL').required().asUrlString();
+
+  const issuer = env.get('OAUTH2_TOKEN_ISSUER').asUrlString();
+  const issuerRegex = env.get('OAUTH2_TOKEN_ISSUER_REGEX').asRegExp('u');
+  if (
+    (issuer === undefined && issuerRegex === undefined) ||
+    (issuer !== undefined && issuerRegex !== undefined)
+  ) {
+    throw new Error('Either OAUTH2_TOKEN_ISSUER or OAUTH2_TOKEN_ISSUER_REGEX should be set');
+  }
+
+  return {
+    audience,
+    domain,
+    issuer: issuer ?? (issuerRegex as any),
```

Comment:
The `as any` is a workaround for https://github.com/nearform/fastify-auth0-verify/pull/279